pkg/sass-parser/lib/src/statement/declaration.test.ts

```diff
@@ -0,0 +1,511 @@
+
+  it('is a variable with --', () => {
+    node.prop = '--foo';
+    expect(node.variable).toBe(true);
+  });
+
+  describe('stringifies', () => {
+    describe('to SCSS', () => {
+      describe('with default raws', () => {
+        it('with no children', () =>
+          expect(new Declaration({prop: 'foo', value: 'bar'}).toString()).toBe(
+            'foo: bar',
+          ));
+
+        it('with a value and children', () =>
+          expect(
+            new Declaration({
+              prop: 'foo',
+              value: 'bar',
+              nodes: [{prop: 'baz', value: 'bang'}],
+            }).toString(),
+          ).toBe('foo: bar {\n    baz: bang\n}'));
+
+        it('with only children', () =>
+          expect(
+            new Declaration({
+              prop: 'foo',
+              nodes: [{prop: 'baz', value: 'bang'}],
+            }).toString(),
+          ).toBe('foo: {\n    baz: bang\n}'));
+      });
+
+      describe('with before', () => {
+        it('on its own', () =>
+          expect(
+            new Declaration({
+              prop: 'foo',
+              value: 'bar',
+              raws: {before: '/**/'},
+            }).toString(),
+          ).toBe('foo: bar'));
+
+        it('as a child', () =>
+          expect(
+            new Rule({
+              selector: 'foo',
+              nodes: [{prop: 'bar', value: 'baz', raws: {before: '/**/'}}],
+            }).toString(),
+          ).toBe('foo {/**/bar: baz\n}'));
+      });
+
+      describe('with between', () => {
+        it('with no children', () =>
+          expect(
+            new Declaration({
+              prop: 'foo',
+              value: 'bar',
+              raws: {between: ':/**/'},
+            }).toString(),
+          ).toBe('foo:/**/bar'));
+
+        it('with a value and children', () =>
+          expect(
+            new Declaration({
+              prop: 'foo',
+              value: 'bar',
+              nodes: [{prop: 'baz', value: 'bang'}],
+              raws: {between: ':/**/'},
+            }).toString(),
+          ).toBe('foo:/**/bar {\n    baz: bang\n}'));
+
+        it('with no value and children', () =>
+          expect(
+            new Declaration({
+              prop: 'foo',
+              nodes: [{prop: 'baz', value: 'bang'}],
+              raws: {between: ':/**/'},
+            }).toString(),
+          ).toBe('foo:/**/ {\n    baz: bang\n}'));
+      });
+
+      it('ignores important', () =>
+        expect(
+          new Declaration({
+            prop: 'foo',
+            value: 'bar !important',
+            raws: {important: '!IMPORTANT'},
+          }).toString(),
+        ).toBe('foo: bar !important'));
+
+      it('ignores value', () =>
+        expect(
+          new Declaration({
+            prop: 'foo',
+            value: 'bar',
+            raws: {value: {value: 'bar', raw: 'BAR'}},
+          }).toString(),
+        ).toBe('foo: bar'));
```

Comment:
I'm slightly lost, why is `!important` and `value` ignored?

Reply:
Because in the Sass AST, those are fully parsed expressions in their own right, and the expression ASTs determine how they're serialized rather than the statement-level raws.